Add reddit support

janaSunrise opened the pull request:
This adds Reddit cog which provides multiple neat commands:
* [x] Memes
* [x] Science
* [x] Technology
* [x] Relationship
* [x] Videos

janaSunrise committed:
add reddit feature

janaSunrise committed:
Edit subreddit Lists

ItsDrike commented on cogs/reddit.py:
```diff
@@ -0,0 +1,441 @@
+        name = random.choice(constants.meme_sublist)
+        subreddit = reddit.subreddit(name)
+
+        hotposts = subreddit.hot(limit=100)
+        postlist = list(hotposts)
```
There is no reason for `hotposts` variable to exist when it's only used to be converted into `postlist`, use the variable as a list right away instead.

**NOTE:** You'll need to change all references to `postlist` to `hotposts`
**APPLIES:** This applies to every occurence like this, lines: 55, 92, 129, 167, 197, 227, 264, 371
```suggestion
        hotposts = list(subreddit.hot(limit=100))
```

ItsDrike commented on cogs/reddit.py:
```diff
@@ -0,0 +1,441 @@
+class Reddit(Cog):
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
```
I'd like to see a `>>reddit` command and all of the other ones here would only act as it's subcommand to avoid too many commands in help as these are basically only different subreddits anyway.

I'd suggest making `>>reddit memes`, etc. and when the main command gets run (`reddit`) as a group itself, it will either print help or print random reddit post from given subreddit

ItsDrike commented on cogs/reddit.py:
```diff
@@ -0,0 +1,441 @@
+        )
+        await ctx.send(embed=embed)
+
+    @command()
+    @is_nsfw()
+    async def nsfw(self, ctx: Context) -> None:
+        name = random.choice(constants.nsfw_sublist)
+        subreddit = reddit.subreddit(name)
+
+        hotposts = subreddit.hot(limit=100)
+        postlist = list(hotposts)
+
+        randompost = random.choice(postlist)
+        embed = Embed(
+            title=randompost.title,
+            url=randompost.url,
+            colour=Color.green()
+        )
+        embed.set_image(url=randompost.url)
+        embed.set_footer(
+            text=f"👍 {randompost.score} | 💬 {len(randompost.comments)} | Powered By HotWired",
+        )
+        embed.set_author(
+            name=f"u/{randompost.author.name}",
+            icon_url=randompost.author.icon_img,
+            url=f"https://www.reddit.com/user/{randompost.author.name}"
+        )
+        embed.add_field(
+            name="SubReddit",
+            value=f"[r/{subreddit}](https://www.reddit.com/r/{subreddit}/)",
+            inline=False
+        )
+        await ctx.send(embed=embed)
```
I'm not a huge fan of this, why should we have NSFW reddit posts in the bot?

```suggestion
```

ItsDrike commented on cogs/utils/constants.py:
```diff
@@ -216,6 +216,164 @@
+    "godpussy",
+    "hairypussy",
+    "breakingtheseal",
+    "SexyTummies",
+    "thick",
+    "SlimThick",
+    "dirtysmall",
+    "cumsluts",
+    "GirlsFinishingTheJob",
+    "impressedbycum"
+]
+
+aww_sublist = [
+    "AnimalsBeingJerks",
+    "NatureIsFuckingLit",
+    "WhatsWrongWithYourDog",
+    "StartledCats",
+    "natureismetal",
+    "guineapigs",
+    "Rabbits"
+]
+
+sci_sublist = [
+    "askscience",
+    "Physics",
+    "shittyrobots",
+    "chemicalreactiongifs",
+    "chemistry",
+    "nasa",
+    "EverythingScience",
+    "spaceporn"
+]
+
+relation_sublist = [
+    "relationship_advice",
+    "Tinder",
+    "dating_advice",
+    "DeadBedrooms",
+    "family"
+]
```
Consider using a dict here instead, with categories specified
For example:
```py
reddit = {
    "memes": [
        "memes",
        "dankmemes",
        ...
    ],
    "...": [...]
}
```

janaSunrise committed:
Edit subreddits list, and change it to dict

janaSunrise committed:
Merge master

janaSunrise committed:
Fix reddit commands

ItsDrike commented on bot/__main__.py:
```diff
@@ -27,7 +27,7 @@
     "bot.cogs.comics",
     "bot.cogs.coding",
-    "bot.cogs.documentation"
+    "bot.cogs.reddit",
```
Why is this removing documentation and replacing it with reddit, reddit should be a new cog but there is no reason to remove documentation cog.

ItsDrike commented on bot/cogs/reddit.py:
```diff
@@ -0,0 +1,310 @@
+        """Reddit commands."""
+        pass
+
+    @reddit.command()
```
```suggestion
    @reddit.command(aliases=["meme"])
```
Add a singular version

ItsDrike commented on bot/cogs/reddit.py:
```diff
@@ -0,0 +1,310 @@
+        if "https://v.redd.it/" in randompost.url or "https://youtube.com/" in randompost.url:
+            await ctx.send(randompost.url)
+
+    @reddit.command()
```
```suggestion
    @reddit.command(aliases="tech")
```

ItsDrike reviewed (changes requested):
A few minor things and this will be ready

janaSunrise committed:
Update bot/cogs/reddit.py

Co-authored-by: ItsDrike <itsdrikeofficial@gmail.com>

janaSunrise committed:
Update bot/cogs/reddit.py

Co-authored-by: ItsDrike <itsdrikeofficial@gmail.com>

janaSunrise committed:
Add docs cog

janaSunrise committed:
Merge branch 'reddit' of https://github.com/The-Codin-Hole/HotWired-Bot into reddit

janaSunrise committed:
Solve errors

ItsDrike commented on bot/cogs/documentation.py:
```diff
@@ -5,6 +5,7 @@
 import aiohttp
 import bleach
 import stackexchange
+from stackexchange import Site
```
This isn't necessary, it only clutters imports, using `stackexchange.Site` is fine

ItsDrike commented on bot/cogs/reddit.py:
```diff
@@ -0,0 +1,310 @@
+from discord.ext.commands import Cog, Bot, Context, is_nsfw, group
+import random
+from bot import config
+import praw
```
```suggestion
from praw import Reddit as RedditAPI
```
We're only using `Reddit` class from `praw` there is no point in importing the whole module.
I'm importing it as `RedditAPI` as `Reddit` name is already taken by the Cog class name, but feel free to use a different name
Note: you'll also need to replace every occurrence of `praw.Reddit` with just `Reddit`

ItsDrike commented on bot/cogs/reddit.py:
```diff
@@ -0,0 +1,310 @@
+        name = random.choice(config.reddit["meme"])
+        subreddit = reddit_client.subreddit(name)
+
+        postlist = list(subreddit.hot(limit=100))
+
+        randompost = random.choice(postlist)
+        embed = await reddit_embed(subreddit, randompost)
```
Follow this spacing style in every group command
```suggestion
        postlist = list(subreddit.hot(limit=100))
        randompost = random.choice(postlist)
        
        embed = await reddit_embed(subreddit, randompost)
```

ItsDrike commented on bot/cogs/reddit.py:
```diff
@@ -0,0 +1,310 @@
+
+    @reddit.command()
+    async def new(self, ctx: Context, subreddit: str) -> None:
+        """sends you the fresh posts from a subreddit"""
```
```suggestion
        """Sends you the fresh posts from given subreddit."""
```

ItsDrike commented on bot/cogs/reddit.py:
```diff
@@ -0,0 +1,310 @@
+        pass
+
+    @reddit.command(aliases=["meme"])
+    async def memes(self, ctx: Context) -> None:
+        name = random.choice(config.reddit["meme"])
```
Add docstring here and to all other subcommands of reddit.

ItsDrike commented on bot/cogs/reddit.py:
```diff
@@ -0,0 +1,310 @@
+
+        if randompost.over_18:
+            if ctx.channel.is_nsfw():
+                if "https://v.redd.it/" in randompost.url:
+                    await ctx.send(randompost.title)
+                    await ctx.send(randompost.url)
+                elif "https://youtube.com/" in randompost.url:
+                    await ctx.send(randompost.title)
+                    await ctx.send(randompost.url)
+                else:
+                    embed = Embed(
+                        title=randompost.title,
+                        url=randompost.url,
+                        colour=Color.green()
+                    )
+                    embed.set_image(url=randompost.url)
+                    embed.set_footer(
+                        text=f"👍 {randompost.score} | 💬 {len(randompost.comments)} | Powered By HotWired",
+                    )
+                    embed.set_author(
+                        name=f"u/{randompost.author.name}",
+                        icon_url=randompost.author.icon_img,
+                        url=f"https://www.reddit.com/user/{randompost.author.name}"
+                    )
+                    embed.add_field(
+                        name="SubReddit",
+                        value=f"[r/{subreddit}](https://www.reddit.com/r/{subreddit}/)",
+                        inline=False
+                    )
```
Shouldn't this be compressed into the `reddit_embed` function?

janaSunrise committed:
Change as requested in PR

ItsDrike commented on bot/config.py:
```diff
@@ -401,6 +233,12 @@
     default_languages = safe_load(file)
 
 
+with open("bot/assets/reddit.json", "r") as f:
+    contents = f.read()
+
+reddit = json.loads(contents)
+
+
```
There is no need to clutter `config.py` with this, you can do this directly in the `reddit` cog

janaSunrise committed:
Replace config to Direct usage

janaSunrise committed:
Clean code

janaSunrise committed:
Reddit > Subreddit

ItsDrike committed:
Fix bugs and handle blank env vars

- Rename `subreddit` to `subreddits` as `subreddit` is used in multiple
  functions where it gets redefined, overriding the `subreddit` variable
- Add a handler for blank enviromental variables for reddit credentials
